feature/src/common/frame.h

```diff
@@ -27,6 +27,8 @@ struct data
     size_t offset;
     FILE *ref_rfile;
     FILE *dis_rfile;
+    int print_progress;
```

Comment:
Not sure if it is best to put `print_progress` in `data`. It doesn't go well with the rest of the fields.

Why do we need to add it as optional in the first place? What was the reason it failed py2.7?

Reply:
Yes, that was what caused the failures.

IMO it does not make a lot of sense for a command-line program to automatically print progress, as it gets “muddled” with other, potentially machine-parsable output (unless progress is sent to STDERR and real info is sent to STDOUT). But this is a preference that some tools follow, some don't.

Reply:
@li-zhi OK. I removed the `--progress` option again. The progress was printed to stdout instead of stderr before, and that seemed to cripple py2.7's parsing code. The tests passed after sending the progress to stderr according to slhck's suggestion.